server.js

```diff
@@ -1,30 +1,62 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import booksData from './data/books_10k.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('To see the collection type in /books')
+})
+
+// Return the whole collection with 10k books
+app.get('/books', (reg, res) => {
+  res.json(booksData)
+
+})
+
+// Return books of given year and query books with average rating over 4
+app.get('/books/year/:year', (reg, res) => {
+  const year = reg.params.year
+  const topRated = reg.query.toprated
+  let booksFromYear = booksData.filter((item) => item.original_publication_year === +year)
+
+  if (topRated) {
+    booksFromYear = booksFromYear.filter((item) => item.average_rating > 4)
+  }
+
+  if (booksFromYear.length === 0) {
+    res.send('No books of that year exists in the collection')
+  }
+  res.json(booksFromYear)
+})
```

Comment:
Endpoints should be named after what they return and not how they are filtered. These could therefore be query params under your first endpoint:
`/books?year=2010&toprated=true`